main.go

```diff
@@ -122,6 +126,17 @@ func main() {
 		os.Exit(1)
 	}
 
+	cfg, err := config.GetConfig()
+	if err != nil {
+		setupLog.Error(err, "")
+		os.Exit(1)
+	}
+	kclient, err := kubernetes.NewForConfig(cfg)
+	if err != nil {
+		setupLog.Error(err, "")
+		os.Exit(1)
+	}
+
 	if err = (&controllers.WatcherAPIReconciler{
```

Comment:
I got  a question about this part (even though this PR does not modify it). Is the order in which the reconcilers are called important? The WatcherAPI one is called before the Watcher one, which seems counterintuitive to me, since the Watcher CR will own WatcherAPI and the others

Reply:
That's just because we created first the watcherapi but I agree in that watcher should go first although tbh, i think the controllers should be able to reconcile in any order.

Reply:
thanks @amoralej I wanted to know if the order of the reconcile mattered, if not, we can address this later

Reply:
it can and does.

but the way we generally address that is that the CRs for the indivugal components of the project are only created after the deps are meant

so we would only create the watcher-api cr after the keystone account and db ectra are successfully reconsiled,

the first step in the reconsile loop of the watcher API controller should also effectively be validating this inputs.
so if we pass the name of a secrete for example then it should assert that it exists and has the required content before proceeding and set input-ready to false if the preconditions are not meant.

in general the reconsiles can happen in any order but for the initial reconsole fo the watcher cr to create each of the others there will be a somewhat fixed ordering based on depenices.

the best practice is to make sure things are event based rather then pooling.

so we avoid waiting for other resource to reconsile and instead exit the reconsile loop with an error and wait to be triggered by a change in one of the resources we watch.